site/public/js/admin-gradeable-updates.js

```diff
@@ -243,43 +243,129 @@ function ajaxCheckBuildStatus() {
-                successCallback(response.data);
+            
+            if(students_lines_present != 1){
+                alert("Cannot Proccess file, requires exactly one labelled 'student' column");
+                return;
             }
-            else if (response.status === 'fail') {
-                errorCallback(response.message, response.data);
+            
+            if(graders_lines_present != 1){
+                alert("Cannot Proccess file, requires exactly one labelled 'grader' column");
+                return;
             }
-            else {
-                alert('Internal server error');
-                console.error(response.message);
+
+            for(var i=1;i<lines.length;i++){
+
+                var built_line = {};
+                var cells=lines[i].split(",");
+
+                for(var j=0;j<cells.length;j++){
+                    if(cells[j].trim() != ''){
+                        built_line[headers[j].trim()]= cells[j].trim();
+                    }
+                }
+                jsonFile[i-1] = built_line;
             }
-        },
-        error: function (response) {
-            setGradeableUpdateComplete();
-            console.error('Failed to parse response from server: ' + response);
+            let container = $('#container-rubric');
+            if (container.length === 0) {
+                alert("UPDATES DISABLED: no 'container-rubric' element!");
+                return;
+            }
+            // Don't process updates until the page is done loading
+            if (!container.is(':visible')) {
+                return;
+            }
+            p_values['peer_graders_list'] = jsonFile;
+            setGradeableUpdateInProgress();
+            $.getJSON({
+                type: "POST",
+                url: buildCourseUrl(['gradeable', gradeable_id, 'update']),
+                data: p_values,
+                success: function (response) {
+                    if (Array.isArray(response['data'])) {
+                        if (response['data'].includes('rebuild_queued')) {
+                            ajaxCheckBuildStatus(gradeable_id,'unknown');
+                        }
+                    }
+                    setGradeableUpdateComplete();
+                    if (response.status === 'success') {
+                        successCallback(response.data);
+                    }
+                    else if (response.status === 'fail') {
+                        errorCallback(response.message, response.data);
+                    }
+                    else {
+                        alert('Internal server error');
+                        console.error(response.message);
+                    }
+                },
+                error: function (response) {
+                    setGradeableUpdateComplete();
+                    console.error('Failed to parse response from server: ' + response);
+                }
+            });            
         }
-    });
+    }
```

Comment:
Can this else and the tail end of the above if be consolidated?

Reply:
I don't think so ,the getJson must be inside the reader.onload = function. I tried making a helper function, but that blew up.